src/main/java/use_case/registration/RegistrationInteractor.java

```diff
@@ -0,0 +1,109 @@
+package use_case.registration;
+
+import entity.User;
+import utility.validations.PasswordValidator;
+import utility.validations.UsernameValidator;
+
+/**
+ * The Registration Interactor.
+ */
+public class RegistrationInteractor implements RegistrationInputBoundary {
+
+    // Reference to the output boundary (presenter) for displaying feedback
+    private final RegistrationOutputBoundary presenter;
+
+    // Reference to the data access object as an interface type
+    private final RegistrationDataAccessInterface dataAccess;
+
+    /**
+     * Constructs a RegistrationInteractor with dependencies on presenter, data access, and user factory.
+     *
+     * @param presenter   The output boundary for displaying results to the user.
+     * @param dataAccess  The data access object for storing and retrieving users.
+     */
+    public RegistrationInteractor(RegistrationOutputBoundary presenter, RegistrationDataAccessInterface dataAccess) {
+        this.presenter = presenter;
+        this.dataAccess = dataAccess;
+    }
+
+    /**
+     * Processes the registration input data to register a new user.
+     * Checks if the username and password are valid and unique. If registration is successful,
+     * it creates and saves a new user and sends success feedback to the presenter.
+     * Otherwise, it sends an error message to the presenter.
+     *
+     * @param inputData The input data containing the username and password for registration.
+     */
+    @Override
+    public void execute(RegistrationInputData inputData) {
+        try {
+            // Check for empty username/password
+            validateInput(inputData);
+
+            // Create a new user
+            User newUser = new User(inputData.username(), inputData.password());
+
```

Comment:
Remember to add initial balance to the user. Create a `static final` constant `INITIAL_BALANCE` in this class and add the balance to new user.